src/sync/chase_lev.rs

```diff
@@ -158,69 +155,93 @@ impl<T> Deque<T> {
     unsafe fn push(&self, data: T) {
         let guard = epoch::pin();
 
-        let mut b = self.bottom.load(SeqCst);
-        let t = self.top.load(SeqCst);
-        let mut a = self.array.load(SeqCst, &guard).unwrap();
-        let size = b - t;
+        let mut b = self.bottom.load(Relaxed);
+        let t = self.top.load(Acquire);
+        let mut a = self.array.load(Relaxed, &guard).unwrap();
+        
+        let size = b.wrapping_sub(t);
```

Comment:
Generally speaking, I wanted to follow more closely to the papers which may have accounted for some integer overflow. The standard Chase-Lev paper has a footnote (4) specifying that they assume top will never overflow, implying that bottom might. In practice, this is probably not a big deal, but we should probably use i64 rather than isize. I don't mind reverting this change if you have a strong opinion.